### PR TITLE
Add resend confirmation for existing account email page to register journey

### PR DESCRIPTION
**Pull request by wilpow-github**
### Context

We need a sign in/registration flow that is independent of DQT and TRNs.

### Changes proposed in this pull request

Add a new page at /sign-in/register/resend-confirm-existing-account-email following the designs of the existing page at  /sign-in/trn/resend-email-confirmation (Note there is no field for the user to change the email address the code is sent to here.).

### Checklist

-   [x] Attach to Trello card
-   [ ] Rebased master
-   [ ] Cleaned commit history
-   [ ] Tested by running locally
-   [ ] Reminder created to manually clean any removed app settings post deployment

### CODE REVIEW

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/BaseExistingEmailPageModel.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                return EmailPinGenerationResult.Failed(new ViewResult()
+                {
+                    StatusCode = 429,
+                    ViewName = "TooManyRequests"
+                });
+
+            case EmailPinGenerationFailedReasons.NonPersonalAddress:
+            case EmailPinGenerationFailedReasons.InvalidAddress:
+                _logger.LogWarning($"Validation failed for existing email: {email} ");
+                return EmailPinGenerationResult.Failed(this.PageWithErrors());
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/BaseExistingEmailPageModel.cs:42]: Not sure what the appropriate response to receiving a validation error for an existing email should be. Note NonPersonalAddress should never occur as we set requiresValidation to false for GenerateEmailPin

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserVerification/PinGenerationFailedReasons.cs [hunk 2]
```diff
@@ -1,9 +1,8 @@
 namespace TeacherIdentity.AuthServer.Services.UserVerification;
 
-[Flags]
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserVerification/PinGenerationFailedReasons.cs:3]: I couldn't see any reason for these to be flags

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/BaseEmailPinGenerationPageModel.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            if (existingUser is null)
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private async Task<EmailPinGenerationFailedReasons> TryGenerateEmailPinForEmail(string email, bool requiresValidation)
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/BaseEmailPinGenerationPageModel.cs:53]: Is `requiresValidation` parameter required here?

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserVerification/EmailPinGenerationFailedReasons.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+namespace TeacherIdentity.AuthServer.Services.UserVerification;
+
+public enum EmailPinGenerationFailedReasons
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserVerification/EmailPinGenerationFailedReasons.cs:3]: If not a Flags enum then singular name is preferred 
```suggestion
public enum EmailPinGenerationFailedReason
```

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserVerification/PinGenerationFailedReasons.cs [hunk 5]
```diff
@@ -1,9 +1,8 @@
...
 public enum PinGenerationFailedReasons
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Services/UserVerification/PinGenerationFailedReasons.cs:3]:
```suggestion
public enum PinGenerationFailedReason
```

**Review by gunndabad — APPROVED** [review]

**Pull request merged**